src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java

```diff
@@ -105,6 +105,38 @@ private void UnaryAndPostfix() {
         }
     }
 
+    public void checkBooleanStatements() {
+        boolean a = true;
+        int b = 42;
+        int c = 42;
+        int d = 32;
+        if((b == c) == a
+                && (( // violation, unnecessary parenthesis, the latter one
+                                (b==c)==(d>=b)==a!=(c==d))
+                || (b<=c)!=a==(c>=d))) {
```

Comment:
your example are over complicated..... ok to keep it. 
but please add several simple examples that easy to consume by people.
please use expression from issue and please take few cases from diff report.

Reply:
Done.